packages/okta-auth-js/lib/browser/browser.js

```diff
@@ -202,8 +203,87 @@ proto.signIn = function (opts) {
   });
 };
 
-proto.signOut = function () {
-  return this.session.close();
+// Ends the current application session, clearing all local tokens
+// Ends the user's Okta session using the API or redirect method
+proto.signOut = function (options) {
+  options = util.extend({}, options);
+
+  // postLogoutRedirectUri must be whitelisted in Okta Admin UI
+  var postLogoutRedirectUri = options.postLogoutRedirectUri || this.options.postLogoutRedirectUri;
+  var redirect = postLogoutRedirectUri || options.redirect;
+  if (!redirect) {
+    // Clear local tokens
+    this.tokenManager.clear();
+
+     // DELETE /api/v1/sessions/me
+    return this.session.close();
+  }
+
+  // will redirect to Okta-hosted issuer page for login, unless redirectUri was specified
+  var issuer = this.options.issuer || this.options.url;
+
+  // Support top-level ORG Authorization Server
+  if (issuer.indexOf('/oauth2') === -1) {
+    issuer = issuer + '/oauth2';
+  }
+
+  // If idToken is not available and no postLogoutRedirectUri is specified, we will redirect here
+  var oktaHome = new URL(issuer).origin;
+
+  // idToken is required to logout from Okta authorization server
+  var idToken = options.idToken;
+
+  // Can specify the window/frame to reload/redirect
+  var _window = options.window || window;
```

Comment:
We don't have to. I put it in primarily for testing, but I did the unit tests in jest so am not actually using the hook. This would be very useful for a test in Karma. (which is the type of case where it might be running in an iframe)

Reply:
If we don't need it I think this should be removed.